app/javascript/pagination.js

```diff
@@ -0,0 +1,22 @@
+
+import { Turbo } from '@hotwired/turbo-rails';
+
+window.addEventListener('turbo:load', async (event) => {
```

Comment:
I would consider using a stimulus controller to for the event handling 

```
export default class extends Controller {
  initialize() {
  }

  submit() {
    this.element.requestSubmit();
  }
}
```
to connect the controller to the drop-down you would add something like this to the `form_with` 
```
<%= form_with(url: articles, method: :get,
        data: {
          turbo_frame: "results",
          turbo_action: "advance",
          controller: "form",
          action: "input->form#submit"
        }) do |f| %>
```

and wrap the part of the page you want to be updated with this 
```
<%= turbo_frame_tag "results" do %>
  Render list here
<% end %>
```